cassandra-protocol/src/frame.rs

```diff
@@ -117,6 +123,85 @@ impl Frame {
         &self.warnings
     }
 
+    /// Parses the raw bytes of a cassandra frame returning a [`Frame`] struct.
+    /// The typical use case is reading from a buffer that may contain 0 or more frames and where the last frame may be incomplete.
+    /// The possible return values are:
+    /// * `Ok(Frame, usize)` - The first frame in the buffer has been succesfully parsed.
+    ///     + [`Frame`] - The parsed frame
+    ///     + `usize` - How many bytes from the buffer have been read.
+    /// * `Err(ParseFrameError::NotEnoughBytes)` - There are not enough bytes to parse a single frame, [`Frame::from_buffer`] should be recalled when it is possible that there are more bytes.
+    /// * `Err(_)` - The frame is malformed and you should close the connection as this method does not provide a way to tell how many bytes to advance the buffer in this case.
+    pub fn from_buffer(
+        data: &[u8],
+        compressor: Compression,
+    ) -> Result<(Frame, usize), ParseFrameError> {
+        if data.len() < HEADER_LEN {
+            return Err(ParseFrameError::NotEnoughBytes);
+        }
+
+        let body_len = try_i32_from_bytes(&data[5..9]).unwrap() as usize;
```

Comment:
Maybe return an error here instead of panicking.

Reply:
An error there is impossible.
a valid i32 can always be constructed from 4 bytes.

Reply:
True. It would be great to avoid going to and from a slice along the way.